streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/StreamToTableJoinScalaIntegrationTestImplicitSerdes.scala

```diff
@@ -101,13 +102,59 @@ class StreamToTableJoinScalaIntegrationTestImplicitSerdes extends JUnitSuite wit
 
     val actualClicksPerRegion: java.util.List[KeyValue[String, Long]] =
       produceNConsume(userClicksTopic, userRegionsTopic, outputTopic)
-
     streams.close()
 
     import collection.JavaConverters._
     assertEquals(actualClicksPerRegion.asScala.sortBy(_.key), expectedClicksPerRegion.sortBy(_.key))
   }
 
+    @Test def testShouldCountClicksPerRegionWithMissingTopic(): Unit = {
```

Comment:
Thanks for adding this integration test! It looks reasonable, but we'd generally adding integration Java test under `org.apache.kafka.streams.integration`, not `org.apache.kafka.streams.scala` (the latter is only for the scala API only).

Reply:
And also the title `testShouldCountClicksPerRegionWithMissingTopic` is confusing, it should be `shouldShutdownWithMissingTopic` right? And note this class is for `StreamToTableJoinScalaIntegrationTestImplicitSerdes`, so not the right class to add this test. I'd suggest adding a new test class under the above mentioned package for this test case, like `AssignmentErrorHandlingIntegrationTest`, and this test case be `shouldAutoShutdownOnIncompleteMetadata`.

Reply:
I took a look at streams/src/test/java/org/apache/kafka/streams/integration/StreamTableJoinIntegrationTest.java where cluster is not involved.
If I move the new test there, a lot of Scala code for setting up the testing environment would be repeated.

It seems more intuitive if the new integration test is added in this test class in terms of code reuse.

Reply:
I understand the code duplication, but still adding a test case that has nothing to do with `StreamToTableJoinScalaIntegrationTestImplicitSerdes` is not recommended. I'd still suggest making a new class and duplicate the setup code a bit.